pkg/generate/app/imageref.go

```diff
@@ -366,11 +367,17 @@ func (r *ImageRef) DeployableContainer() (container *kapi.Container, triggers []
 				},
 			},
 		}
+		// If the image is coming from an image stream, don't use the image stream name as an image.
+		// This will allow upstream deployment controller to pause the deployment until triggers supply
+		// the resolved image.
+		if isDeployment {
```

Comment:
@tnozicka @smarterclayton is it possible to create DC today with image field `" "`? If not, we should allow it, so we are in sync with upstream and don't need this `if`. It will also mean we won't generate DC with broken image fields after all ;-)

Reply:
I think we use the same validation as upstream as that is per pod template in general and " " is valid in DC